acceptance/tests/partitions/partitions_test.go

```diff
@@ -786,170 +359,280 @@ func TestPartitionsWithMesh(t *testing.T) {
+				k8s.DeployKustomize(t, clientClusterStaticServerOpts, cfg.NoCleanupOnFailure, cfg.DebugDirectory, "../fixtures/cases/static-server-inject")
+				if c.destinationNamespace == defaultNamespace {
+					k8s.DeployKustomize(t, clientClusterStaticClientOpts, cfg.NoCleanupOnFailure, cfg.DebugDirectory, "../fixtures/cases/static-client-partitions/default-ns-default-partition")
+				} else {
+					k8s.DeployKustomize(t, clientClusterStaticClientOpts, cfg.NoCleanupOnFailure, cfg.DebugDirectory, "../fixtures/cases/static-client-partitions/ns-default-partition")
+				}
+				// Check that both static-server and static-client have been injected and now have 2 containers in server cluster.
+				for _, labelSelector := range []string{"app=static-server", "app=static-client"} {
+					podList, err := serverClusterContext.KubernetesClient(t).CoreV1().Pods(metav1.NamespaceAll).List(context.Background(), metav1.ListOptions{
+						LabelSelector: labelSelector,
+					})
+					require.NoError(t, err)
+					require.Len(t, podList.Items, 1)
+					require.Len(t, podList.Items[0].Spec.Containers, 2)
+				}
+
+				// Check that both static-server and static-client have been injected and now have 2 containers in client cluster.
+				for _, labelSelector := range []string{"app=static-server", "app=static-client"} {
+					podList, err := clientClusterContext.KubernetesClient(t).CoreV1().Pods(metav1.NamespaceAll).List(context.Background(), metav1.ListOptions{
+						LabelSelector: labelSelector,
+					})
+					require.NoError(t, err)
+					require.Len(t, podList.Items, 1)
+					require.Len(t, podList.Items[0].Spec.Containers, 2)
+				}
+
+				// Make sure that services are registered in the correct namespace.
+				// If mirroring is enabled, we expect services to be registered in the
+				// Consul namespace with the same name as their source
+				// Kubernetes namespace.
+				// If a single destination namespace is set, we expect all services
+				// to be registered in that destination Consul namespace.
+				// Server cluster.
+				// We are going to test that static-clients deployed in each partition can
+				// access the static-servers running in another partition.
+				// ie default -> secondary and secondary -> default.
+				services, _, err := consulClient.Catalog().Service(staticServerName, "", serverQueryServerOpts)
+				require.NoError(t, err)
+				require.Len(t, services, 1)
+
+				services, _, err = consulClient.Catalog().Service(staticClientName, "", clientQueryServerOpts)
+				require.NoError(t, err)
+				require.Len(t, services, 1)
+
+				// Client cluster.
+				services, _, err = consulClient.Catalog().Service(staticServerName, "", serverQueryClientOpts)
+				require.NoError(t, err)
+				require.Len(t, services, 1)
+
+				services, _, err = consulClient.Catalog().Service(staticClientName, "", clientQueryClientOpts)
+				require.NoError(t, err)
+				require.Len(t, services, 1)
+
+				defaultPartitionExports := &api.PartitionExportsConfigEntry{
```

Comment:
could we make it a CRD and put it in the fixture? I think this should be the workflow that we think users on k8s will use.

Reply:
good call. yeah ill create a CRD fixture for this!